code/network/multi_pxo.cpp

```diff
@@ -1585,34 +1587,71 @@ void multi_pxo_strip_space(char *string1,char *string2)
 	}
 }
 
-// fire up the given URL
-void multi_pxo_url(char * /*url*/)
+static int open_url(const char *url)
 {
-#if 0
-	// execute the shell command
-	int r = (int) ShellExecute(NULL, NOX("open"), url, NULL, NULL, SW_SHOW);
-	if (r < 32) {		
-		switch (r) {
-			case 0:	
-			case ERROR_BAD_FORMAT: 
-			case SE_ERR_ACCESSDENIED: 
-			case SE_ERR_ASSOCINCOMPLETE: 
-			case SE_ERR_DDEBUSY:
-			case SE_ERR_DDEFAIL:
-			case SE_ERR_DDETIMEOUT:
-			case SE_ERR_DLLNOTFOUND:
-			case SE_ERR_OOM:
-			case SE_ERR_SHARE:			
-			case SE_ERR_NOASSOC:
-			case ERROR_FILE_NOT_FOUND:
-			case ERROR_PATH_NOT_FOUND:
-				popup(PF_USE_AFFIRMATIVE_ICON | PF_TITLE_RED | PF_TITLE_BIG,1,POPUP_OK,XSTR("Warning\nCould not locate/launch default Internet Browser",943));
-				break;
-		}					
+#if defined(_WIN32) || defined(__APPLE__)
+	const char *open_cmd = "open";
+#else
+	const char *open_cmd = "xdg-open";
+#endif
+
+	char s_url[256];
+
+	// make sure it's a valid web uri
```

Comment:
Typo uri instead of url?

Reply:
The idea is to make sure we are passing a valid URI scheme to launch a web browser but not check that it's actually a valid URL that the browser can do something with. A basic security check so we aren't executing random commands but passing the bigger validation and security checks off to the browser.

So in context "uri" should be more appropriate than "url".